src/deployment/npm/jobs/internal-public-npm-package.job.yaml

```diff
@@ -3,7 +3,7 @@ jobs:
   - deployment: Deploy
     displayName: Deploy
     environment: NPM Internal
-    condition: and(succeeded(), and(not(eq(variables['Build.Reason'], 'PullRequest')), not(eq(variables['Build.Reason'], 'Schedule'))))
+    condition: and(eq(variables['Build.ShouldPublish'], 'true'), and(succeeded(), and(not(eq(variables['Build.Reason'], 'PullRequest')), not(eq(variables['Build.Reason'], 'Schedule')))))
```

Comment:
I don't think this will be able to access the `ShouldPublish` variable here, because you are in a different job and a different stage. I think you need to use dependencies for this with output variables.

So I think the condition will need something like this in `stageDependencies.Build.<JobId>.outputs['UpdateVersions.ShouldPublish']`. I'm not exactly sure what the `JobId` would be as one is not explicitly set, and I'm also not sure if `UpdateVersions` is correct either. May be ones we have to try out to know